### PR TITLE
[SPARK-6602][Core] Remove some places in core that calling SparkEnv.actorSystem

### PR DESCRIPTION
**Pull request by zsxwing**

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala [hunk 1]
```diff
@@ -115,11 +116,9 @@ private[spark] class CoarseMesosSchedulerBackend(
     }
     val command = CommandInfo.newBuilder()
       .setEnvironment(environment)
-    val driverUrl = AkkaUtils.address(
-      AkkaUtils.protocol(sc.env.actorSystem),
+    val driverUrl = sc.env.rpcEnv.uriOf(
       SparkEnv.driverActorSystemName,
-      conf.get("spark.driver.host"),
-      conf.get("spark.driver.port"),
+      RpcAddress(conf.get("spark.driver.host"), conf.get("spark.driver.port").toInt),
```
Comment (rxin) [inline core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala]: i noticed that AkkaUtils.address's "port" is Any type. Do you know why?

Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala]: No idea. I will change it to `Int`.

**Comment by zsxwing** [comment]
cc @andrewor14 for the yarn changes

File: yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala [hunk 2]
```diff
@@ -231,8 +231,14 @@ private[spark] class ApplicationMaster(
...
         .getOrElse("")
 
-    allocator = client.register(yarnConf,
-      if (sc != null) sc.getConf else sparkConf,
+    val _sparkConf = if (sc != null) sc.getConf else sparkConf
+    val driverUrl = _rpcEnv.uriOf(
+        SparkEnv.driverActorSystemName,
+        RpcAddress(_sparkConf.get("spark.driver.host"), _sparkConf.get("spark.driver.port").toInt),
+        CoarseGrainedSchedulerBackend.ENDPOINT_NAME)
+    allocator = client.register(driverUrl,
```
Comment (andrewor14) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala]: `driverUrl` should technically go onto the next line

**Comment by andrewor14** [comment]
Both core and YARN parts LGTM. I'm merging this into master.

**Pull request closed without merging**